http-server/http-server.go

```diff
@@ -3,34 +3,72 @@ package httpserver
 import (
 	"bytes"
 	"context"
+	"crypto/subtle"
+	"crypto/x509"
 	"encoding/json"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
 	"net/http"
+	"os"
 	"runtime"
 	"time"
 
+	grpcRuntime "github.com/grpc-ecosystem/grpc-gateway/runtime"
 	"github.com/packethost/pkg/log"
 	"github.com/pkg/errors"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/tinkerbell/tink/protos/hardware"
+	"github.com/tinkerbell/tink/protos/template"
+	"github.com/tinkerbell/tink/protos/workflow"
 )
 
 var (
 	gitRev         = "unknown"
 	gitRevJSON     []byte
+	grpcListenAddr = "localhost:42113"
```

Comment:
this port number should be in a config file, not hard coded  here.

Reply:
How should the config file look like? Would it suffice if I just have the port be read from an environment variable and set a default port for when the variable unset?

Reply:
Yes, there shall be a default value if there is no configured value. If we have quite some values, a file will be easy to manage.

Reply:
I think the closest to the config file you were referring to is the [`docker-compose.yml`](https://github.com/tinkerbell/tink/blob/c49b6c2dec7df60a2588c18617ea2e72c3aeb68d/deploy/docker-compose.yml#L18) that is currently used to run all of the services.  
I have added two new environment variables to allow the user to specify their own ports.

Please refer to this commit here: https://github.com/tinkerbell/tink/pull/64/commits/aee3b9f33e1ddae86c91db0999c9b3fc022bbf0e